nltk/tokenize/ssp.py

```diff
@@ -0,0 +1,142 @@
+# Natural Language Toolkit: Tokenizers
+#
+# Copyright (C) 2001-2019 NLTK Project
+# Author: Christopher Hench <chris.l.hench@gmail.com>
+#         Alex Estes
+# URL: <http://nltk.sourceforge.net>
+# For license information, see LICENSE.TXT
+#
+# Reference: Otto Jespersen, Lehrbuch der Phonetik.
+#            University of Toronto (Leipzig, Teubner, 1904).
+
+"""
+The Sonority Sequencing Principle is a language agnostic algorithm developed
+by Otto Jesperson in 1904. The sonorous quality of a phoneme is judged by the
+openness of the lips. Syllable breaks occur before troughs in sonority.
+
+The default implementation uses the English alphabet, but the `sonority_hiearchy`
+can be modified to IPA or any other alphabet for the use-case.
+
+Importantly, if a custom hiearchy is supplied and vowels span across more than
+one level, they should be given separately to the `vowels` attribute.
+"""
+
+from __future__ import unicode_literals
+from nltk.tokenize.api import TokenizerI
+
+
+class SonoritySequencingPrinciple(TokenizerI):
+    '''
+    Syllabifies words based on the Sonority Sequencing Principle (SSP)
+
+    >>> SSP = SonoritySequencingPrinciple()
+    >>> SSP.tokenize('justification')
+    ['jus', 'ti', 'fi', 'ca', 'tion']
+    '''
+
+    def __init__(self, sonority_hierarchy=False):
```

Comment:
Lets specify the language here `def __init__(self, lang='en', snority_hierarchy=False):`